src/Streamly/Internal/Data/Fold.hs

```diff
@@ -1722,16 +1744,7 @@ demuxWithDefault_ f kv (Fold dstep dinitial dextract) =
                                              dacc
                                              (Set.insert k bst)
                                              (Map.delete k mp)
-                            Done1 _ ->
-                                if Map.size mp == 1
-                                -- XXX This should just be Done?
-                                then Done1 ()
-                                else Partial
-                                       $ DemuxingWithDefault
-                                             dacc
-                                             (Set.insert k bst)
-                                             (Map.delete k mp)
-        else return $ Done1 ()
+        else return $ Done ()
```

Comment:
We are losing the input here, we should not run into this case.

Reply:
Yeah, I wanted to return an `error` here. This was on my scratchpad. Good catch.